mhm/__init__.py

```diff
@@ -0,0 +1,52 @@
+import os
+import json
+import pandas as pd
+
+
+PARAMS_INDIVIDUAL = 'params_individual.json'
+PARAMS_MODEL = 'params_model.json'
+
+
+def read_json_as_dict(fpath: str):
+    """Read content of a JSON file as Python dictionary.
+
+    Args:
+        fpath (str): file path.
+
+    Returns:
+        dict: result dictionary.
+    """
+    assert os.path.isfile(fpath), 'File not found: %s.' % fpath
+    with open(fpath) as file:
+        data = json.load(file)
+    return data
+
+def create_empty_hypotheses(dir_params: str):
+    """Create empty hypotheses files in a user-defined folder.
+
+    Args:
+        dir_params (str): folder that contains the agent and model parameter files.
+    """
+    fpath_params_individual = os.path.join(dir_params, PARAMS_INDIVIDUAL)
+    fpath_params_model = os.path.join(dir_params, PARAMS_MODEL)
+    params_individual = read_json_as_dict(fpath_params_individual) 
+    params_model = read_json_as_dict(fpath_params_model)
+    
+    actions = params_model['actions']
+    lockdown_policies = params_model['lockdown_policies']
+    status = params_model['status']
+    columns = ['actions', 'baseline']
+    for key, value in params_individual.items():
+        # categorical columns will be one-hot encoded by value
```

Comment:
I understand what you tried to do here. However, now the matrices do not match in the number/type of columns anymore. One of the differences I was able to spot is that in the output parameters  `gender` (which is a binary variable) it's split into "gender_M" and "gender_F" -- this is not necessary as the variable gender is supposed to be binary, and so I wonder how can we improve on this here? I see that you have `if isinstance(value[0][0], str):" but indeed gender should return `FALSE`. Ah! I see now that in the individual.json you have 

`
"gender":[
        ["M", "F"],
        [0.5, 0.5]    
`

Which indeed is not binary. OK. I'll try to change it back to binary to see if I can fix this problem.

Reply:
I can explain it. My understanding is that `gender` is indeed binary but a not boolean feature. The point here is that hypotheses on gender "0" (or False) will be automatically ignored when doing dot production (0 times anything is 0).

Reply:
Ehy @jiqicn -- I am finally done reviewing this PR!
So I think it's great, thank you very much.

I looked again at the gender variable, and while I think it's possible to make explicit the hypotheses for both M and F even when there is a binary variable, if this didn't seem obvious to you, I think it will not be obvious to the LA as well. Also, it's much simpler from a user point of view to specify explicitly M and F. So I agree that having M/F is the best way to go here.

I am now dealing though with a mismatch in the number and type of columns contained in `action_effects_on_mh.csv` and `action_effects_on_contacts.csv` and the lockdown files. If I work with the files that I made previously (that are with gender 0 and 1) they work fine, but if I replace them with the ones that you made with M/F they don't work anymore, and I am not sure why. I did not manage to find the mismatch yet. At this point I think it's best if we start writing tests so that we don't have any problems. I think we have enough code now to make this step necessary.

So In the end, I am going to approve this PR and merge it to main (many thanks again for this brilliant work), but I will reinstate the old csv files, in the meantime that I figure out why your files don't work and how to make sure that this is fixed if it happens in the future with tests, and by raising an error in our code if the input indeed has a mismatch.

Looking also at how things are organised, I believe also we should revise the idea of having a json file for specifying the steps, length and type of lockdown per step, I mean the `params_model.json`. I think it makes much more sense if the user gets this type of file as output (i.e., to save the parameters of that specific simulation), rather than as input -- Indeed at the moment I am not using it. However, I don't have a strong opinion on this and we can discuss this in person.

many thanks again and have a nice weekend :wink: